### PR TITLE
A0-2377: emergency finalizer script

### PR DESCRIPTION
**Pull request by DamianStraszak**
Added a small helper program that allows to run the emergency finalization procedure. For usage and details see https://www.notion.so/cardinalcryptography/Emergency-Finalizer-701a27d81d7e49faa0144ca7f6d43d52

### CODE REVIEW

File: aleph-client/src/pallets/aleph.rs [hunk 1]
```diff
@@ -124,7 +124,7 @@ impl<C: ConnectionApi> AlephRpc for C {
         let raw_signature: &[u8] = signature.as_ref();
         let params = rpc_params![raw_signature, hash, number];
 
-        let _: () = self.rpc_call(method.to_string(), params).await?;
```
Comment (kostekIV) [inline aleph-client/src/pallets/aleph.rs:127]: thx for fixing this...

**Review by mike1729 — APPROVED** [review]

**Review by deuszx — COMMENTED** [review]
What if network gets stuck and nodes finalized different branches?

File: bin/finalizer/seed.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+//Alice
```
[the remark this answers is not in the document]
Comment (timorl) [reply bin/finalizer/seed.txt:1]: Asking partially, because this looks as if it was put here to bait someone to commit an actual key to the repo. ;p

**Review by timorl — APPROVED** [review]
Looks good, one nit and one question, neither should block merging.

File: bin/finalizer/src/commands.rs [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+    primary: HashNum,
+    secondaries: BTreeMap<String, HashNum>,
+}
+
+struct FinalizationPlan {
+    finalized_base: HashNum,
+    target: HashNum,
+}
+
+impl fmt::Display for ChainStatus {
```
[the remark this answers is not in the document]
Comment (DamianStraszak) [reply bin/finalizer/src/commands.rs]: You are right, fixed.

**Pull request merged**